### PR TITLE
No rule check for non-existing super implementor

### LINKED ISSUE #12766: No rule check for non-existing super implementor
**Bug description**
We do not have a code rule check that will inform about non-existing implementor of a message send if the message is sent by `super`.

**To Reproduce**
1. in any class, create this method
```smalltalk
uniqueMessage 

	self asString.
	^ super uniqueMessage
```
2. no warning

(asString message is here just to prevent warning about a method using super send only)

### PR DESCRIPTION
**Pull request by astares**
Fix #12766

### CODE REVIEW

**Comment by astares** [comment]
@pavel-krivanek can you please review?

image

**Comment by MarcusDenker** [comment]
- [CI] Windows testrunner has problems #12763

File: src/GeneralRules/ReSentButNotUnderstoodBySuperRule.class.st [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+]
+
+{ #category : #running }
+ReSentButNotUnderstoodBySuperRule >> check: aMethod forCritiquesDo: aCriticBlock [
+
+	| superclass problemSends |
+	superclass := aMethod methodClass superclass.
+
+	problemSends := aMethod sendNodes select: [ :msgSend | msgSend receiver isSuperVariable ].
+	problemSends := problemSends reject: [ :msgSend |
```
Comment (demarey) [inline src/GeneralRules/ReSentButNotUnderstoodBySuperRule.class.st:30]: I would rename `problemSends` to `notUnderstoodSends`

**Review by demarey — COMMENTED** [review]
Thanks for the PR.
It would be really nice to add some basic tests for this new rule

File: src/GeneralRules/ReSentButNotUnderstoodBySuperRule.class.st [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"This number should be unique and should change only when the rule completely change semantics"
+
+	^'SentButNotUnderstoodBySuperRule'
+]
+
+{ #category : #running }
+ReSentButNotUnderstoodBySuperRule >> check: aMethod forCritiquesDo: aCriticBlock [
+
+	| superclass problemSends |
+	superclass := aMethod methodClass superclass.
```
Comment (pavel-krivanek) [inline src/GeneralRules/ReSentButNotUnderstoodBySuperRule.class.st:27]: superclass can be `nil`

**Review by pavel-krivanek — COMMENTED** [review]
thanks!

**Comment by guillep** [comment]
So, to summarise:
 - this PR needs to introduce tests
 - the new rule needs to handle the case of nil superclasses

**Comment by astares** [comment]
We close this one in favour of https://github.com/pharo-project/pharo/pull/15705

**Pull request closed without merging**